### PR TITLE
CLI improvements

### PR DESCRIPTION
**Pull request by thetristan**
Replaces #16.

Same changes, but based on the branch at my fork.

### CODE REVIEW

File: flexiconf-cli/src/main/scala/se/blea/flexiconf/cli/CLI.scala [hunk 1]
```diff
@@ -1,174 +1,62 @@
...
-      |                   args: <configPath> <schemaPath>""".stripMargin
+  def usage = {
+    val padLen = 2 + actions.map(_.name)
+      .maxBy(_.length)
+      .length
 
-  def main(_args: Array[String]): Unit = {
-    val optionalArgs = _args.toList.takeWhile(_.startsWith("-"))
-    implicit val args: List[String] = _args.toList.dropWhile(_.startsWith("-"))
+    val actionUsage = actions.map { act =>
```
Comment (dustyburwell) [inline flexiconf-cli/src/main/scala/se/blea/flexiconf/cli/CLI.scala]: This isn't something necessary today, but perhaps consider bringing in a dependency like [Scopt](https://github.com/scopt/scopt) to handle the option parsing.

**Comment by dustyburwell** [comment]
This looked really familiar since I had previously checked it out in the last PR. LGTM

**Comment by thetristan** [comment]
Rebased against latest master.

**Comment by dustyburwell** [comment]
Great, thanks! I was about to do that :P

**Pull request merged**